packages/core/src/baseclient.ts

```diff
@@ -94,25 +95,20 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
    * @param transport The (old) Transport instance for the client to use (TODO(v7): remove)
    * @param newTransport The NewTransport instance for the client to use
    */
-  protected constructor(options: O, transport: Transport, newTransport?: NewTransport) {
+  protected constructor(options: O) {
     this._options = options;
 
+    let url;
     if (options.dsn) {
       this._dsn = makeDsn(options.dsn);
+      // TODO(v7): Figure out what to do with transport when dsn is not defined
+      const api = initAPIDetails(this._dsn, options._metadata, options.tunnel);
+      url = getEnvelopeEndpointWithUrlEncodedAuth(api.dsn, api.tunnel);
     } else {
       IS_DEBUG_BUILD && logger.warn('No DSN provided, client will not do anything.');
     }
 
-    // TODO(v7): remove old transport
-    this._transport = transport;
-    this._newTransport = newTransport;
-
-    // TODO(v7): refactor this to keep metadata/api outside of transport. This hack is used to
-    //           satisfy tests until we move to NewTransport where we have to revisit this.
-    (this._transport as unknown as { _api: Partial<APIDetails> })._api = {
-      ...((this._transport as unknown as { _api: Partial<APIDetails> })._api || {}),
-      metadata: options._metadata || {},
-    };
+    this._transport = options.transport({ ...options.transportOptions, url: url || '' });
```

Comment:
this is the key here, making sure the client still controls what is being passed in as options.